src/legendhpges/base.py

```diff
@@ -105,6 +110,65 @@ def _decode_polycone_coord(self) -> tuple[list[float], list[float]]:
         Must be overloaded by derived classes.
         """
 
+    def distance_to_surface(
+        self, coords: np.ndarray | list, surface_indices: list | None = None
+    ) -> np.ndarray:
+        """Compute the distance of a set of points to the nearest detector surface.
+
+        Parameters
+        ----------
+        coords
+            2D array of `(x,y,z)` coordinates for each point, first index corresponds to the point, second to the dimension `(x,y,z)`.
+        surface_indices
+            list of indices of surfaces to consider. If `None` (the default) all surfaces used.
```

Comment:
I don't understand what is the shape of `coords` from this description. Also what is the type of the elements in `surface_indices`? Mention or add reference to another function